auth/src/main/java/com/firebase/ui/auth/util/smartlock/SignInDelegate.java

```diff
@@ -0,0 +1,403 @@
+                    // Smart lock selector cancelled, go to the AuthMethodPicker screen
+                    startAuthMethodChoice();
+                } else if (resultCode == RESULT_FIRST_USER) {
+                    // TODO: (serikb) figure out flow
+                }
+                break;
+            case RC_IDP_SIGNIN:
+            case RC_AUTH_METHOD_PICKER:
+            case RC_EMAIL_FLOW:
+                finish(resultCode, new Intent());
+                break;
+            case RC_PLAY_SERVICES:
+                if (resultCode != RESULT_OK) {
+                    finish(resultCode, new Intent());
+                }
+                break;
+        }
+    }
+
+    private String getEmailFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getId();
+    }
+
+    private String getAccountTypeFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getAccountType();
+    }
+
+    private String getPasswordFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getPassword();
+    }
+
+    private void handleCredential(Credential credential) {
+        mCredential = credential;
+
+        if (IdentityProviders.GOOGLE.equals(credential.getAccountType())) {
+            // Google account, rebuild GoogleApiClient to set account name and then try
+            initGoogleApiClient(credential.getId());
+            // Try silent sign-in with Google Sign In API
+            Auth.GoogleSignInApi.silentSignIn(mGoogleApiClient);
+        } else {
+            // Email/password account
+            String status = String.format("Signed in as %s", credential.getId());
+            Log.d(TAG, status);
+        }
+    }
+
+    private void initGoogleApiClient(String accountName) {
+        GoogleSignInOptions.Builder gsoBuilder = new GoogleSignInOptions
+                .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail();
+
+        if (accountName != null) {
+            gsoBuilder.setAccountName(accountName);
+        }
+
+        if (mGoogleApiClient != null) {
+            mGoogleApiClient.disconnect();
+        }
+
+        mGoogleApiClient = new GoogleApiClient.Builder(getContext())
+                .addConnectionCallbacks(this)
+                .addApi(Auth.CREDENTIALS_API)
+                .addApi(Auth.GOOGLE_SIGN_IN_API, gsoBuilder.build())
+                .build();
+        mGoogleApiClient.connect();
+    }
+
+    private void startAuthMethodChoice() {
+        List<IDPProviderParcel> providers = mFlowParams.providerInfo;
+
+        // If the only provider is Email, immediately launch the email flow. Otherwise, launch
+        // the auth method picker screen.
+        if (providers.size() == 1
+                && providers.get(0).getProviderType().equals(EmailAuthProvider.PROVIDER_ID)) {
```

Comment:
While refactoring `IDPSignInContainerActivity`, I'll change this to be if any provider is the only one selected we'll launch that directly. Right now, if only Google (for example) is selected, there will be one lonely button in `AuthMethodPickerActivity`. Is that a reasonable change or was there a reason for only caring about whether or not the email provider was the only one selected?